components/script/dom/gamepad.rs

```diff
@@ -143,10 +194,44 @@ impl Gamepad {
         self.index.set(index);
     }
 
+    pub fn update_timestamp(&self, timestamp: f64) {
+        self.timestamp.set(timestamp);
+    }
+
     pub fn notify_event(&self, event_type: GamepadEventType) {
         let event = GamepadEvent::new_with_type(&self.global(), event_type, &self);
         event
             .upcast::<Event>()
             .fire(self.global().as_window().upcast::<EventTarget>());
     }
+
+    // https://www.w3.org/TR/gamepad/#dfn-initializing-axes
+    // https://www.w3.org/TR/gamepad/#fingerprinting-mitigation
+    pub fn init_axes(&self) {
+        // Initialize the number of axes in the "standard" gamepad mapping.
+        // See above comment on buttons
+        let initial_axes: Vec<f64> = vec![
+            0., // Horizontal axis for left stick (negative left/positive right)
+            0., // Vertical axis for left stick (negative up/positive down)
+            0., // Horizontal axis for right stick (negative left/positive right)
+            0., // Vertical axis for right stick (negative up/positive down)
+        ];
+        self.axes
+            .set_data(GlobalScope::get_cx(), &initial_axes)
+            .expect("Failed to set axes data on gamepad.")
+    }
+
+    #[allow(unsafe_code)]
+    pub fn update_axis(&self, axis_index: usize, value: f32) {
+        let mut axis_vec = self.axes.get_internal().unwrap();
+        unsafe {
+            axis_vec.as_mut_slice()[axis_index] = value as f64;
+        }
+    }
+
+    pub fn update_button(&self, button_index: usize, pressed: bool, touched: bool, value: f32) {
```

Comment:
For this and other methods, instead of adding a comment at the call site, it would be better to have a name that matches the spec(here `map_and_normalize_buttons`), and add a doc comment with the link <https://w3c.github.io/gamepad/#dfn-map-and-normalize-buttons>.

A comment at the call site then becomes unnecessary.

Reply:
Done